src/Nest/QueryDsl/TermLevel/Range/RangeQueryJsonConverter.cs

```diff
@@ -26,28 +26,47 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 			var jo = firstProp.Value.Value<JObject>();
 			if (jo == null) return null;
 
+			var fq = GetRangeQuery(serializer, jo);
 
-			var isNumeric = !jo.Properties().Any(p=>p.Name == "format" || p.Name == "time_zone")
-				&& jo.Properties().Any(p=> _rangeKeys.Contains(p.Name) && (p.Value.Type  == JTokenType.Integer || p.Value.Type == JTokenType.Float));
+			fq.Name = GetPropValue<string>(jo, "_name");
+			fq.Boost = GetPropValue<double?>(jo, "boost");
+			fq.Field = field;
 
+			return fq;
+		}
+
+		private static IRangeQuery GetRangeQuery(JsonSerializer serializer, JObject jo)
+		{
+			var nameIsValid = !jo.Properties().Any(p => p.Name == "format" || p.Name == "time_zone");
+
+			var isLong = nameIsValid && CheckType(jo, JTokenType.Integer);
+			var isNumeric = nameIsValid && CheckType(jo, JTokenType.Float);
```

Comment:
I think this can be simplified further. 

For `isNumeric` and DateRangeQuery cases, `CheckType` will end up being called twice. It looks like it should be possible to enumerate/iterate over `jo.Properties()` once and determine from the existence of "format", "time_zone" or token type, which `IRangeQuery` it is. I was thinking something like

```c#
private static IRangeQuery GetRangeQuery(JsonSerializer serializer, JObject jo)
{
	foreach (var property in jo.Properties())
	{
		if (property.Name == "format" || property.Name == "time_zone")
			return FromJson.ReadAs<DateRangeQuery>(jo.CreateReader(), serializer);

		if (_rangeKeys.Contains(property.Name))
		{
			if (property.Value.Type == JTokenType.Integer)
				return FromJson.ReadAs<LongRangeQuery>(jo.CreateReader(), serializer);					
			if (property.Value.Type == JTokenType.Float)
				return FromJson.ReadAs<NumericRangeQuery>(jo.CreateReader(), serializer);
			
			return FromJson.ReadAs<DateRangeQuery>(jo.CreateReader(), serializer);
		}
	}
}
```

what do you think?

Reply:
Yes, thank you. LGTM. "format" and "time_zone" checks must be done for all properties in jo. When we check for a type, we can not immediately return the result, because LongRange is used, if there was no property of type float, but there was a property of type int. However, we can iterate once and I think that the code will looks better.